### PR TITLE
DATAUP-451 null inputs

### PR DESCRIPTION
**Pull request by briehl**
# Description of PR purpose/changes

This addresses 2 bugs.
1. Inputs that are flagged as optional with an empty value by default weren't properly being accepted. This was fixed by using the processed spec object's version of the default values instead of what's in the default spec. I.e. an empty string isn't appropriate for a float input, so that gets cast to a null. The problem was, as it was before, that an empty string was being put in there instead of a null.
2. Once that was fixed, I found that on cell initialization, by default, the cell should be ready to run. But it wasn't! It required the user to click through each file type in the configure tab first, which would update the ready state for each of those. This is fixed by having the bulk import cell investigate its parameter state at cell startup, instead of waiting on the configure tab.

I was having some strange, seemingly unrelated, issues when working locally. Such as cells not registering their delete event, or even running `setupCell`. I'm not sure if that's related to the changes, or to my browser being silly. So I'm running tests here.

# Jira Ticket / Issue #
https://kbase-jira.atlassian.net/browse/DATAUP-451
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: nbextensions/bulkImportCell/bulkImportCell.js [hunk 1]
```diff
@@ -571,37 +580,55 @@ define([
...
+                        ['editingComplete', 'editingIncomplete'].includes(curState.state)
+                    ) {
+                        updateEditingState();
+                    } else {
+                        updateState();
+                    }
+                    cell.renderMinMax();
+                    // // force toolbar refresh
+                    // // eslint-disable-next-line no-self-assign
+                    // cell.metadata = cell.metadata;
```
Comment (ialarmedalien) [inline nbextensions/bulkImportCell/bulkImportCell.js:629]: rm commented-out code

File: test/unit/spec/nbextensions/bulkImportCell/bulkImportCell-spec.js [hunk 2]
```diff
@@ -272,6 +273,32 @@ define([
...
+            );
+            await TestUtil.waitForElementState(runButton, () => {
+                return (
+                    !runButton.classList.contains('hidden') &&
+                    !runButton.classList.contains('disabled') &&
+                    resetButton.classList.contains('hidden')
+                );
+            });
+            expect(cell.metadata.kbase.bulkImportCell.state.state).toBe('editingComplete');
+            // cellWidget.deleteCell();
```
Comment (ialarmedalien) [inline test/unit/spec/nbextensions/bulkImportCell/bulkImportCell-spec.js:299]: rm commented-out code

File: test/unit/spec/nbextensions/bulkImportCell/bulkImportCell-spec.js [hunk 3]
```diff
@@ -272,6 +273,32 @@ define([
             });
         });
 
+        it('Should start up in "editingComplete" state when initialized with proper data', async () => {
+            const cell = Mocks.buildMockCell('code');
+            Jupyter.notebook = Mocks.buildMockNotebook();
+            const cellWidget = BulkImportCell.make({
```
Comment (ialarmedalien) [inline test/unit/spec/nbextensions/bulkImportCell/bulkImportCell-spec.js:279]: do you need the `cellWidget` variable or can you just run `BulkImportCell.make({..})`?

Comment (briehl) [reply test/unit/spec/nbextensions/bulkImportCell/bulkImportCell-spec.js]: Nope, I was experimenting with deleting the cell (and stopping widgets and such) at the end. That, however, probably means removing / canceling running Promises, and making sure that the `make` function returns as a Promise anyway. I just forgot to undo that part.

**Review by ialarmedalien — APPROVED** [review]
👍

**Pull request merged**